swan-cern/scripts/before-notebook.d/01_rucio.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# Author: Giovanni Guerrieri, Enrique Garcia Garcia (2025)
+# Copyright CERN
+# This script configures the environment for the usage 
+# of the Rucio JupyterLab extension.
+
+# Check if the RUCIO_JUPYTERLAB_VERSION environment variable is set.
+if [[ -z "$RUCIO_JUPYTERLAB_VERSION" ]]; then
+    _log "RUCIO_JUPYTERLAB_VERSION is not set. Please set it to the desired version."
+    exit 1
+fi
+
+# Verify if the specified Rucio JupyterLab version exists in the CVMFS directory.
+if [[ ! -d "/cvmfs/sw.escape.eu/rucio-jupyterlab/${RUCIO_JUPYTERLAB_VERSION}" ]]; then
+    _log "Rucio JupyterLab version ${RUCIO_JUPYTERLAB_VERSION} is not available."
+    exit 1
+fi
+
+_log "Setting up Rucio JupyterLab environment for version ${RUCIO_JUPYTERLAB_VERSION}..."
+
+# Define the base path for the ESCAPE CVMFS repository.
+ESCAPE_CVMFS_PATH="/cvmfs/sw.escape.eu"
+
+# Define the path to the specific Rucio JupyterLab version within the CVMFS repository.
+RUCIO_CVMFS_PATH="${ESCAPE_CVMFS_PATH}/rucio-jupyterlab/${RUCIO_JUPYTERLAB_VERSION}"
+
+# Set the RUCIO_HOME environment variable to point to the Rucio JupyterLab installation directory.
+export RUCIO_HOME="${RUCIO_CVMFS_PATH}"
+
+# Add the Rucio JupyterLab binary directory to the PATH environment variable.
+export USER_PATH="${RUCIO_CVMFS_PATH}/bin"
+export PATH="${USER_PATH}:${PATH}"
+
+# Add the Rucio JupyterLab Python packages directory to the PYTHONPATH environment variable.
+export USER_PYTHONPATH="${RUCIO_CVMFS_PATH}/lib/python3.11/site-packages"
```

Comment:
This hardcoding of `python3.11` could perhaps be resolved by adding a script (e.g. `setup.sh`) to `${RUCIO_CVMFS_PATH}` that we source and gets us a variable that points to the site-packages dir? This way, the Rucio dir on CVMFS could be updated with newer Python versions without us having to update the image.

Reply:
Actually, note that LCG 108 (now default stack in SWAN) has moved already to Python 3.12.